### PR TITLE
chore(arch-364): add surge demo

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

Github action only cover test and lint

**What is the chosen solution to this problem?**

Add surge support (demo, umd)

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Review by tlnd-mhuchet — APPROVED** [review]

File: .github/workflows/pr-demo.yml [hunk 1]
```diff
@@ -0,0 +1,142 @@
+name: PR demo CI
+
+on:
+  pull_request:
+    branches: [master]
```
Comment (frassinier) [inline .github/workflows/pr-demo.yml]:
```suggestion
```

File: .github/workflows/pr-demo.yml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        run: |
+          yarn install --frozen-lockfile
+          yarn global add surge
+          echo "$(yarn global bin)" >> $GITHUB_PATH
+          yarn prepublishOnly
+          yarn test:demo --quiet
+          cd packages/cmf-router && yarn install && yarn prepublishOnly && cd ../..
+          cd packages/dataviz && yarn install && yarn test:demo --quiet && cd ../..
+          echo prepare demo folder
+          rm -rf .static
```
Comment (frassinier) [inline .github/workflows/pr-demo.yml:46]: It will not be there
```suggestion
```

File: .github/workflows/pr-demo.yml [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      - name: Install and build
+        run: |
+          yarn install --frozen-lockfile
+          yarn global add surge
+          echo "$(yarn global bin)" >> $GITHUB_PATH
+          yarn prepublishOnly
+          yarn test:demo --quiet
+          cd packages/cmf-router && yarn install && yarn prepublishOnly && cd ../..
+          cd packages/dataviz && yarn install && yarn test:demo --quiet && cd ../..
+          echo prepare demo folder
```
Comment (frassinier) [inline .github/workflows/pr-demo.yml:45]:
```suggestion
          echo "Prepare demo folder"
```

File: .github/workflows/pr-demo.yml [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+          mkdir -p ".static/${{ github.event.number }}"
+          cp -R .static/@talend ".static/${{ github.event.number }}"
+
+
+      - name: deploy if master
+        if: github.ref == 'refs/heads/master'
+        run: surge --project .static --domain "talend.surge.sh"
+        env:
+          SURGE_LOGIN: frontend@talend.com
```
Comment (frassinier) [inline .github/workflows/pr-demo.yml:123]: Should be a secret?

**Review by frassinier — APPROVED** [review]

**Pull request merged**